src/Orchard.Cms.Web/Modules/Orchard.OpenId/Views/Admin/Edit.cshtml

```diff
@@ -132,145 +118,147 @@
+            else {
+                clientSecretHints.each(function () { this.innerText = this.innerText.replace("client_id, client_secret, ", "client_id, "); });
+            }
         }
-    }
 
 
-    $("#AllowRefreshTokenFlow").change(function () {
-        refreshOfflineAccessTip();
-    });
+        $("#AllowRefreshTokenFlow").change(function () {
+            refreshOfflineAccessTip();
+        });
 
-    function refreshOfflineAccessTip(defaultValue) {
-        var offlineAccessHints = $("#AllowPasswordFlowRecommendedHint, #AllowAuthorizationCodeFlowRecommendedHint, #AllowHybridFlowRecommendedHint");
-        var allowRefreshTokenFlow = $("#AllowRefreshTokenFlow");
-        if (defaultValue === allowRefreshTokenFlow.prop('checked'))
-            return;
+        function refreshOfflineAccessTip(defaultValue) {
+            var offlineAccessHints = $("#AllowPasswordFlowRecommendedHint, #AllowAuthorizationCodeFlowRecommendedHint, #AllowHybridFlowRecommendedHint");
+            var allowRefreshTokenFlow = $("#AllowRefreshTokenFlow");
+            if (defaultValue === allowRefreshTokenFlow.prop('checked'))
+                return;
 
-        if (allowRefreshTokenFlow.prop('checked')) {
-            offlineAccessHints.each(function () { this.innerText = this.innerText.replace("roles", "roles, offline_access"); });
-        }
-        else {
-            offlineAccessHints.each(function () { this.innerText = this.innerText.replace(", offline_access", ""); });
+            if (allowRefreshTokenFlow.prop('checked')) {
+                offlineAccessHints.each(function () { this.innerText = this.innerText.replace("roles", "roles, offline_access"); });
+            }
+            else {
+                offlineAccessHints.each(function () { this.innerText = this.innerText.replace(", offline_access", ""); });
+            }
         }
-    }
-
-    $("#AllowClientCredentialsFlow, #AllowPasswordFlow, #AllowAuthorizationCodeFlow, #AllowHybridFlow, #AllowImplicitFlow, #AllowRefreshTokenFlow").change(function () {
-        refreshFlows();
-    });
 
-    function refreshFlows() {
-        refreshRoleGroup();
-        refreshAllowRefreshTokenFlowVisibility();
-        refreshRedirectSettings();
-    }
-    function refreshRoleGroup() {
-        $("#RoleGroup").collapse($("#AllowClientCredentialsFlow").prop('checked') ? "show" : "hide");
-    }
-    function refreshAllowRefreshTokenFlowVisibility() {
-        var allowRefreshTokenFlow = $("#AllowRefreshTokenFlow");
+        $("#AllowClientCredentialsFlow, #AllowPasswordFlow, #AllowAuthorizationCodeFlow, #AllowHybridFlow, #AllowImplicitFlow, #AllowRefreshTokenFlow").change(function () {
+            refreshFlows();
+        });
 
-        if (($("#AllowPasswordFlow").prop('checked') || $("#AllowAuthorizationCodeFlow").prop('checked') || $("#AllowHybridFlow").prop('checked'))) {
-            allowRefreshTokenFlow.removeAttr("disabled");
+        function refreshFlows() {
+            refreshRoleGroup();
```

Comment:
You removed the ability of setting roles to a client app authenticated through Client Credentials, so this call and the called method should be removed.
However, the important thing is: why did you remove that feature? Did you enabled an alternate way of granting permissions to an app when Client Credentials is enabled?